src/Adapter/Order/AbstractOrderHandler.php

```diff
@@ -343,4 +346,40 @@ protected function updateOrderDetailsWithSameProduct(
             $orderDetail->update();
         }
     }
+
+    /**
+     * @param Order $order
+     * @param Cart $cart
+     *
+     * @throws PrestaShopException
+     * @throws PrestaShopDatabaseException
+     */
+    protected function updateOrderDetailsTax(Order $order, Cart $cart): void
+    {
+        // Fields that are updated (refunds are not modified, even though the tax changed we need to keep track
+        // of the actual refunded amount).
+        $updatedFields = [
+            'unit_price_tax',
+            'total_price_tax',
+            'total_shipping_price_tax',
+        ];
+        $taxAddressId = $order->{Configuration::get('PS_TAX_ADDRESS_TYPE', null, null, $order->id_shop)};
+        $taxAddress = new Address($taxAddressId);
+        $computingPrecision = $this->getPrecisionFromCart($cart);
+
+        foreach ($order->getOrderDetailList() as $row) {
+            $orderDetail = new OrderDetail($row['id_order_detail']);
+            $orderDetail->id_tax_rules_group = (int) Product::getIdTaxRulesGroupByIdProduct($orderDetail->product_id, Context::getContext());
```

Comment:
```suggestion
        $context = Context::getContext();

        foreach ($order->getOrderDetailList() as $row) {
            $orderDetail = new OrderDetail($row['id_order_detail']);
            $orderDetail->id_tax_rules_group = (int) Product::getIdTaxRulesGroupByIdProduct($orderDetail->product_id, $context);
```

Reply:
It's not more performant since it's a singleton but ok 😉